File: 12_circle/circle.jl
function area(r::Float64)
    return π * r^2
end

function radius_from_area(A::Float64)
    return sqrt(A / π)
end

function get_value()
    print("Enter a value for the measurement: ")
    val = readline()
    return parse(Float64, val)
end

function output_measures(A, r, d)
    println("Area: $(A)")
    println("Radius: $(r)")
    println("Diameter: $(d)")
end

print("What measurement will you input? (area/radius/diameter) ")
measure = readline()

if measure == "area"
    A = get_value()
    r = radius_from_area(A)
    d = r * 2
    output_measures(A, r, d)
elseif measure == "radius"
    r = get_value()
    A = area(r)
    d = r * 2
    output_measures(A, r, d)
elseif measure == "diameter"
    d = get_value()
    r = d / 2
    A = area(r)
    output_measures(A, r, d)
else
    print("Not a supported input measurement.")
end

File: 12_circle/circle.cpp
#include <iostream>
#include <cmath>

using namespace std;

const double PI = 3.14;

double area(double r) {
    return PI * pow(r, 2);
};

double radius_from_area(double a) {
    return sqrt(a / PI);
};

void output_measures(double a, double r, double d) {
    cout << "Area: " << a << endl;
    cout << "Radius: " << r << endl;
    cout << "Diameter: " << d << endl;
};

double get_value() {
    cout << "Input value for the measurement: ";
    double val;
    cin >> val;
    
    return val;
};

int main() {

    cout << "What measurement will you be inputing? (area/diameter/radius) ";
    string measure;
    cin >> measure;
    double val;
    double a, r, d;

    if (measure == "area") {
        val = get_value();
        r = radius_from_area(val);
        d = 2 * r;
        output_measures(val, r, d);
    } else if (measure == "radius") {
        val = get_value();
        a = area(val);
        d = val * 2;
        output_measures(a, val, d);
    } else if (measure == "diameter") {
        val = get_value();
        r = val / 2;
        a = area(r);
        output_measures(a, r, val);
    } else {
        cout << "Not a supported input measure.";
    };

     return 0;
};